core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java

```diff
@@ -332,16 +332,22 @@ private void mountDisks(StackView stack, Map<String, String> candidateAddresses,
         }
     }
 
-    public void updateClusterConfigs(@Nonnull StackDto stackDto) {
+    public void updateClusterConfigs(@Nonnull StackDto stackDto, boolean bootstrap) {
         try {
             Set<Node> allNodes = stackUtil.collectNodes(stackDto, emptySet());
             Set<Node> reachableNodes = stackUtil.collectReachableNodesByInstanceStates(stackDto);
             List<GatewayConfig> gatewayConfigs = gatewayConfigService.getAllGatewayConfigs(stackDto);
             List<GrainProperties> grainsProperties = grainPropertiesService.createGrainProperties(gatewayConfigs, stackDto, reachableNodes);
             SaltConfig saltConfig = createSaltConfig(stackDto, grainsProperties);
             ExitCriteriaModel exitCriteriaModel = clusterDeletionBasedModel(stackDto.getStack().getId(), stackDto.getCluster().getId());
+            if (bootstrap) {
+                modifyStartupMountRole(stackDto, reachableNodes, GrainOperation.ADD);
+            }
             hostOrchestrator.initSaltConfig(stackDto, gatewayConfigs, allNodes, saltConfig, exitCriteriaModel);
             hostOrchestrator.runService(gatewayConfigs, reachableNodes, saltConfig, exitCriteriaModel);
+            if (bootstrap) {
+                modifyStartupMountRole(stackDto, reachableNodes, GrainOperation.REMOVE);
+            }
```

Comment:
What about the `mountDisks()` call? Is not it needed? (Just trying to guess from what is done in `executeRunClusterServices()`.)

Reply:
mountDisks is not needed. All i need is that the mount-instance-storage file on the nodes to be updated with the right values, and when restarted the node will mount the disks in the right mount path using the file. modifyStartupMountRole(stackDto, reachableNodes, GrainOperation.ADD) will make sure that the updated salt configs are cascaded from gateway nodes to the group nodes.